### PR TITLE
feat(FOROME-1405): create dashboard widget component

### PR DESCRIPTION
**Pull request by Lapk1n**
[Task](https://quantori.atlassian.net/browse/FOROME-1405)
PR into separate [dashboard branch](https://github.com/ForomePlatform/Anfisa-React-Client/tree/feat/FOROME-1398)

### CODE REVIEW

**Comment by AndreyUstyumenko** [comment]
don't forget to add stories

File: src/pages/filter/common/dashboard/components/body/widget-tab/components/widget-sub-tab/components/widget-sub-tab-enum.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import { Fragment, ReactElement } from 'react'
+
+import { t } from '@i18n'
+import { IWidgetSubTabEnumProps } from '@pages/filter/common/dashboard/dashboard.interfaces'
+
+export const WidgetSubTabEnum = ({
+  unit,
+}: IWidgetSubTabEnumProps): ReactElement => {
+  const variantsLeft = unit.variants ? unit.variants.length - 40 : null
```
Comment (AndreyUstyumenko) [inline src/pages/filter/common/dashboard/components/body/widget-tab/components/widget-sub-tab/components/widget-sub-tab-enum.tsx]: move 40 to a const

**Review by AndreyUstyumenko — APPROVED** [review]

File: src/pages/filter/common/dashboard/components/body/widget-tab/components/widget-sub-tab/components/widget-sub-tab-enum.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+}: IWidgetSubTabEnumProps): ReactElement => {
+  const variantsLeft = unit.variants
+    ? unit.variants.length - MAX_SUB_TAB_ROWS_AMOUNT
+    : null
+
+  return (
+    <>
+      {unit.variants?.map(([variantName, variantValue]: any, index: number) => (
+        <Fragment key={variantName + variantValue}>
+          {index < 41 && (
```
Comment (QSdmitrioul) [inline src/pages/filter/common/dashboard/components/body/widget-tab/components/widget-sub-tab/components/widget-sub-tab-enum.tsx]: Maybe move it to constants?

File: src/pages/filter/common/dashboard/dashboard.tsx [hunk 3]
```diff
@@ -13,16 +13,32 @@ export const Dashboard = ({
   functionalUnits,
   isFetching,
 }: IDashboardProps): ReactElement => {
-  const extenderGroups = dashboardStore.geExtendedGroups(
+  const extendedGroups = dashboardStore.geExtendedGroups(
     groups,
     functionalUnits,
   )
 
+  const [filterValue, setFilterValue] = useState('')
```
Comment (QSdmitrioul) [inline src/pages/filter/common/dashboard/dashboard.tsx]: add type

File: src/pages/filter/common/dashboard/dashboard.utils.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    h: group.units.length + 1,
+  }
+
+  clonedLayout.push(newLayout)
+
+  return clonedLayout
+}
+
+export const getLayoutOnTabHeightChange = (
+  props: IGetLayoutOnHeightChange,
```
Comment (QSdmitrioul) [inline src/pages/filter/common/dashboard/dashboard.utils.tsx:68]: deconstruct props here

**Review by QSdmitrioul — APPROVED** [review]

**Pull request merged**